tests/libvmi/factory.go

```diff
@@ -34,6 +34,18 @@ const (
 // NewFedora instantiates a new Fedora based VMI configuration,
 // building its extra properties based on the specified With* options.
 func NewFedora(opts ...Option) *kvirtv1.VirtualMachineInstance {
+	return newFedora(cd.ContainerDiskFedora, opts...)
+}
+
+// NewCustomFedora instantiates a new Fedora based VMI configuration,
+// building its extra properties based on the specified With* options.
+func NewSriovFedora(opts ...Option) *kvirtv1.VirtualMachineInstance {
```

Comment:
Can you call it `NewExtendedFedora`? It should have been generalized as such in the first place IMO.
We can use it in other places, not only in an SRIOV context.

Reply:
Looks like is more than just GA and friends it really contains SRIOV specific bits:
```
 Fedora cloud edition image with contains all necessary configuration and drivers for sriov lane tests.
    - This image contained the packages:                                        
        - kernel-modules (includes sriov drivers)                               
        - qemu-guest-agent                                                      
    - Configurations:                                                           
        - Enable and start qemu-guest-agent                                     
        - Load kernel modules needed for sriov                                  
          mlx4, mlx5, i40e, igb                          
```

We can create a follow up PR with fedra + GA + whatever we feel is needed and use it on `NewExtendedFedor`

Reply:
It has just additional drivers, how is that a problem?
The current Fedora has much more installations we do not need, so a few more drivers should not matter here.

I do not think we have interest in maintaining multiple images when we do not need to.

Reply:
I prefer to do all or nothing, meaning a followup PR renaming both image and factory name, mixing both is missleading.